File: backend/app/schemas/analytics.py
from pydantic import BaseModel, Field


class DecisionTimeBucketResponse(BaseModel):
    label: str
    seconds: float = Field(ge=0)
    accepted: int = Field(ge=0)
    passed: int = Field(ge=0)
    saved: int = Field(ge=0)


class OpportunityAnalyticsResponse(BaseModel):
    opportunityId: str
    impressions: int = Field(ge=0)
    uniqueViews: int = Field(ge=0)
    acceptedCount: int = Field(ge=0)
    passedCount: int = Field(ge=0)
    savedCount: int = Field(ge=0)
    submissionsCount: int = Field(ge=0)
    acceptanceRate: float = Field(ge=0, le=1)
    averageDecisionTimeMs: int = Field(ge=0)
    medianDecisionTimeMs: int = Field(ge=0)
    averageWatchTimeMs: int = Field(ge=0)
    completionRate: float = Field(ge=0, le=1)
    decisionTimeDistribution: list[DecisionTimeBucketResponse]
    insight: str

File: src/domain/schemas/analytics.ts
import { z } from "zod";

export const decisionTimeBucketSchema = z.object({
  label: z.string(),
  seconds: z.number().nonnegative(),
  accepted: z.number().int().nonnegative(),
  passed: z.number().int().nonnegative(),
  saved: z.number().int().nonnegative(),
});

export const opportunityAnalyticsSchema = z.object({
  opportunityId: z.string(),
  impressions: z.number().int().nonnegative(),
  uniqueViews: z.number().int().nonnegative(),
  acceptedCount: z.number().int().nonnegative(),
  passedCount: z.number().int().nonnegative(),
  savedCount: z.number().int().nonnegative(),
  submissionsCount: z.number().int().nonnegative(),
  acceptanceRate: z.number().min(0).max(1),
  averageDecisionTimeMs: z.number().int().nonnegative(),
  medianDecisionTimeMs: z.number().int().nonnegative(),
  averageWatchTimeMs: z.number().int().nonnegative(),
  completionRate: z.number().min(0).max(1),
  decisionTimeDistribution: z.array(decisionTimeBucketSchema),
  insight: z.string(),
});

export type DecisionTimeBucket = z.infer<typeof decisionTimeBucketSchema>;
export type OpportunityAnalytics = z.infer<typeof opportunityAnalyticsSchema>;
